src/quml-to-odk/quml-to-odk.service.ts

```diff
@@ -0,0 +1,404 @@
+import { Injectable } from '@nestjs/common';
+import { GenerateFormDto } from './dto/generate-form.dto';
+import { HttpService } from '@nestjs/axios';
+import { lastValueFrom, map } from 'rxjs';
+import { exec } from 'child_process';
+import * as XLSX from 'xlsx';
+
+@Injectable()
+export class QumlToOdkService {
+  // config file for the variables being used across the service
+  private config = {
+    questionBankUrl: process.env.QUML_ODK_QUESTION_BANK_URL,
+    questionDetailsUrl: process.env.QUML_ODK_QUESTION_BANK_DETAILS_URL,
+    xlsxFilesPath: process.env.QUML_XLSX_FILE_STORAGE_PATH,
+    odkFormsPath: process.env.QUML_ODK_FORM_FILE_STORAGE_PATH,
+  };
+
+  constructor(private readonly httpService: HttpService) {}
+
+  public async generate(filters: GenerateFormDto) {
+    const questions = await this.fetchQuestions(filters);
+    if (questions.result && questions.result.count) {
+      const parsedResult = this.parseQuestionsForXslxForm(questions, filters);
+      const xlsxFormFile = this.createXslx(
+        parsedResult[0],
+        parsedResult[1],
+        parsedResult[2],
+        parsedResult[3],
+      );
+
+      const odkFormName = +new Date() + '.xml';
+      const odkFormFile = this.config.odkFormsPath + '/' + odkFormName;
+      await this.convertExcelToOdkForm(xlsxFormFile, odkFormFile);
+      return {
+        xlsxFile: xlsxFormFile,
+        odkFile: odkFormFile,
+      };
+    }
+    console.log(
+      'Please ensure there are questions available for the matching combination',
+    );
+    return 'Please ensure there are questions available for the matching combination';
+  }
+
+  private async fetchQuestions(filters: GenerateFormDto): Promise<any> {
+    const requestBody = {
+      request: {
+        filters: {
+          se_boards: filters.boards,
+          gradeLevel: filters.grades,
+          subject: filters.subjects,
+          qType: 'MCQ', // we'll always fetch MCQ questions
```

Comment:
Make this a config. We will expand to questions other than MCQ as well.

Reply:
Done. Reference commit: https://github.com/Samagra-Development/quml2xform/pull/1/commits/cd3caad3799af0c85e39301f07f8d50e75d9755c